RecoTracker/MkFitCMS/src/MkSeedPartitioners-phase2.cc

```diff
@@ -0,0 +1,251 @@
+
+    const float tec_z_extra = 0.0f;  // 10.0f;
+
+    const int size = in_seeds.size();
+
+    auto barrel_pos_check = [](const Track &S, float maxR, float rin, float zmax) -> bool {
+      bool inside = maxR > rin && S.zAtR(rin) < zmax;
+      return inside;
+    };
+
+    auto barrel_neg_check = [](const Track &S, float maxR, float rin, float zmin) -> bool {
+      bool inside = maxR > rin && S.zAtR(rin) > zmin;
+      return inside;
+    };
+
+    auto endcap_pos_check = [](const Track &S, float maxR, float rout, float rin, float zmin) -> bool {
+      bool inside = maxR > rout ? S.zAtR(rout) > zmin : (maxR > rin && S.zAtR(maxR) > zmin);
+      return inside;
+    };
+
+    auto endcap_neg_check = [](const Track &S, float maxR, float rout, float rin, float zmax) -> bool {
+      bool inside = maxR > rout ? S.zAtR(rout) < zmax : (maxR > rin && S.zAtR(maxR) < zmax);
+      return inside;
+    };
+
+    for (int i = 0; i < size; ++i) {
+      const Track &S = in_seeds[i];
+
+      HitOnTrack hot = S.getLastHitOnTrack();
+      float eta = eoh[hot.layer].refHit(hot.index).eta();
+
+      // Region to be defined by propagation / intersection tests
+      TrackerInfo::EtaRegion reg;
+
+      const bool z_dir_pos = S.pz() > 0;
+      const float maxR = S.maxReachRadius();
+
+      if (z_dir_pos) {
+        bool in_tec_as_brl = barrel_pos_check(S, maxR, tecp2_rin, tecp2_zmax);
+
+        if (!in_tec_as_brl) {
+          reg = TrackerInfo::Reg_Endcap_Pos;
+        } else {
+          bool in_tec = endcap_pos_check(S, maxR, tecp1_rout, tecp1_rin, tecp1_zmin - tec_z_extra);
+
+          if (!in_tec) {
+            reg = TrackerInfo::Reg_Barrel;
+          } else {
+            reg = TrackerInfo::Reg_Transition_Pos;
+          }
+        }
+      } else {
+        bool in_tec_as_brl = barrel_neg_check(S, maxR, tecn2_rin, tecn2_zmin);
+
+        if (!in_tec_as_brl) {
+          reg = TrackerInfo::Reg_Endcap_Neg;
+        } else {
+          bool in_tec = endcap_neg_check(S, maxR, tecn1_rout, tecn1_rin, tecn1_zmax + tec_z_extra);
+
+          if (!in_tec) {
+            reg = TrackerInfo::Reg_Barrel;
+          } else {
+            reg = TrackerInfo::Reg_Transition_Neg;
+          }
+        }
+      }
+
+      part.m_region[i] = reg;
+      if (part.m_phi_eta_foo)
+        part.m_phi_eta_foo(eoh[hot.layer].refHit(hot.index).phi(), eta);
+    }
+  }
+
+  [[maybe_unused]] void partitionSeeds1debug(const TrackerInfo &trk_info,
+                                             const TrackVec &in_seeds,
+                                             const EventOfHits &eoh,
+                                             IterationSeedPartition &part) {
+    // Seeds are placed into eta regions and sorted on region + eta.
+
+    // Merge mono and stereo limits for relevant layers / parameters.
+    // TrackerInfo could hold joint limits for sub-detectors.
+    const auto &L = trk_info;
+    const float tecp1_rin = std::min(L[28].rin(), L[29].rin());
```

Comment:
There are a lot of magic numbers here, should they go into a configuration or DB?

Reply:
There is a strong preference to keep the mkfit code independent of the CMS infrastructure code. So, a DB/ES here is not an option; at least not something that can be considered on a medium term.

Similar to what was done in the phase-1 case, it's practical though to define values like `    constexpr int tecp1l_id = 28; tecp1u_id = 29;` for lower/upper parts